pkg/mysqlcluster/volumes.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 Platform9, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysqlcluster
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	api "github.com/presslabs/mysql-operator/pkg/apis/mysql/v1alpha1"
+)
+
+// CleanupVolumeClaims deletes persistent volume claims of mysql cluster if needed
+func CleanupVolumeClaims(client *kubernetes.Interface, cluster *api.MysqlCluster) error {
+	cleanupGracePeriod := cluster.Spec.VolumeSpec.CleanupGracePeriod
+	if cleanupGracePeriod <= 0 {
+		return nil
+	}
+
+	timer := time.NewTimer(time.Duration(cleanupGracePeriod) * time.Second)
+
+	go func(client *kubernetes.Interface, cluster *api.MysqlCluster) {
```

Comment:
It is a problem with this approach, if mysql-operator process dies before timer expires then the PVC will never be deleted.